pkg/deploy/controller/deployment/controller.go

```diff
@@ -324,6 +324,17 @@ func (c *DeploymentController) makeDeployerPod(deployment *kapi.ReplicationContr
 			Labels: map[string]string{
 				deployapi.DeployerPodForDeploymentLabel: deployment.Name,
 			},
+			// Set the owner reference to current deployment, so in case the deployment fails
+			// and the deployer pod is preserved when a revisionHistory limit is reached and the
+			// deployment is removed, we also remove the deployer pod with it.
+			OwnerReferences: []kapi.OwnerReference{{
+				// FIXME: This will have to point to apps.openshift.io/v1 after we switch to
+				// clientsets.
+				APIVersion: "v1",
```

Comment:
@kargakis PTAL again, this has to be set to real version (`deployment.APIVersion` is internal so no version...).